### PR TITLE
Support loading run config to BenchmarkSuite

### PR DESCRIPTION
**Pull request by pzread**
Adds `load_from_run_configs` to `BenchmarkSuite` to load the exported benchmark run configs from e2e test framework.

This change also adds `get_module_path` to `iree_artifacts`, which is a simple way to get the compiled module path without building the whole structure of the artifact directory. I'm thinking the `iree_artifacts.ArtifactsRoot` (which is to help get all paths in the artifact directory) might be over-complicated and can be replaced by simple helper functions (like the `get_moule_path` in this change). Will clean up that part if eventually we find such design is unnecessary.

### CODE REVIEW

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/benchmarks/common/benchmark_suite.py [hunk 1]
```diff
@@ -53,17 +56,41 @@ class BenchmarkCase:
...
-  benchmark_case_dir: str
   benchmark_tool_name: str
+  benchmark_case_dir: Optional[str] = None
+  run_config: Optional[iree_definitions.E2EModelRunConfig] = None
+
+
+def _find_driver_info_by_execution_config(
+    module_execution_config: iree_definitions.ModuleExecutionConfig
+) -> Optional[DriverInfo]:
+  """Finds the matched driver info by the module exeuction config.
```
[the remark this answers is not in the document]
Comment (pzread) [reply build_tools/benchmarks/common/benchmark_suite.py]: Done.

File: build_tools/benchmarks/collect_compilation_statistics.py [hunk 2]
```diff
@@ -10,6 +10,12 @@
 IREE_ENABLE_COMPILATION_BENCHMARKS.
 """
 
+import pathlib
+import sys
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent / ".." / "python"))
```
[the remark this answers is not in the document]
Comment (pzread) [reply build_tools/benchmarks/collect_compilation_statistics.py]: SG, thanks. Done.

File: build_tools/benchmarks/collect_compilation_statistics.py [hunk 3]
```diff
@@ -162,16 +168,20 @@ def main(args: argparse.Namespace):
...
         category=category)
     for benchmark_case in benchmark_cases:
-      flag_file_path = os.path.join(benchmark_case.benchmark_case_dir,
-                                    BENCHMARK_FLAGFILE)
+      benchmark_case_dir = benchmark_case.benchmark_case_dir
+      # TODO(#11076): Support run_config.
+      if benchmark_case_dir is None:
+        raise ValueError("benchmark_case_dir can't be None.")
+
+      flag_file_path = os.path.join(benchmark_case_dir, BENCHMARK_FLAGFILE)
```
[the remark this answers is not in the document]
Comment (pzread) [reply build_tools/benchmarks/collect_compilation_statistics.py]: Benchmark tools have many places using legacy string paths. We'll need to clean up later (#11055).

Changed os.path to pathlib on the touched code in this PR.

File: build_tools/benchmarks/common/benchmark_suite.py [hunk 4]
```diff
@@ -53,17 +56,41 @@ class BenchmarkCase:
...
+
+  Args:
+    module_execution_config: module execution config to match.
+  Returns:
+    A matched driver info. None if not found.
+  """
+  for value in IREE_DRIVERS_INFOS.values():
```
[the remark this answers is not in the document]
Comment (pzread) [reply build_tools/benchmarks/common/benchmark_suite.py]: This is due to the incompatibility between the legacy benchmark suites and the new one. Eventually we will just drop the `DriverInfo` and only rely on `E2EModelRunConfig`.

Changed to a map from `(driver, loader)` to the key of driver info with comments.

**Pull request merged**